slsim/Sources/Supernovae/supernovae_pop.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import scipy.integrate as integrate
+import scipy.interpolate as interp
+
+"""References:
+Oguri and Marshall 2010
+"""
+
+
+def calculate_star_formation_rate(z):
+    """Calculates the cosmic star formation rate. (Eq 13 - Oguri and Marshall 2010)
+
+    :param z: redshift
+    :return: cosmic star formation rate in [(h)yr^(-1)Mpc^(-3)]
+    """
+    if z >= 0:
+        star_formation_rate = (0.0118 + 0.08 * z) / (1 + (z / 3.3) ** 5.2)
+        return star_formation_rate
+    else:
+        return np.nan
+
+
+def t_d_power(t_d):
+    """Calculates the power law constraint on time delay. (Eq 14 - Oguri and Marshall 2010)
```

Comment:
I would phrase it more general as the delay time distribution, and then stating that it is described as a power-law